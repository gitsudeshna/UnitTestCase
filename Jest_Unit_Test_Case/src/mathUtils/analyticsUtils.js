export function track(name) {
    doSomeFireBaseLogic(name);
}