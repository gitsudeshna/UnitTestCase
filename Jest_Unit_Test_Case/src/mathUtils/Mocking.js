import { track } from "./analyticsUtils";

export function addTwoNumbers(x, y) {
    track('addNumberstwo');
    return x + y;
}