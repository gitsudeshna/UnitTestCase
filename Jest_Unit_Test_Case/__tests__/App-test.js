import React from "react";
import 'react-native';
import renderer from 'react-test-renderer';
import { addNumber } from "../src/mathUtils/mathUtils";

it('SnapShot', () => {
  const snap = renderer.create(<addNumber/>).toJSON();
  expect(snap).toMatchSnapshot();
})

test('Add Two Number', () => {
  expect(addNumber(1,2)).toEqual(3);    //toBetruthy, toBeFalsy, toBeDefined, not.toBeDefined()
})
