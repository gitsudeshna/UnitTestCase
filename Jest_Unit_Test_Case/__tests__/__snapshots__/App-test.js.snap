// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot 1`] = `<addNumber />`;
